packages/ilmomasiina-backend/src/routes/signup/updateSignup.ts

```diff
@@ -50,26 +47,38 @@ export default async (
       throw new Forbidden('Signups closed for this event.');
     }
 
+    /** Is this signup already confirmed (i.e. is this the first update for this signup) */
+    const notConfirmedYet = !signup.confirmedAt;
     const questions = event.questions!;
 
     // Check that required common fields are present (if first time confirming)
-    if (!signup.confirmedAt) {
-      if (event.nameQuestion) {
-        if (!data.firstName) {
-          throw new BadRequest('Missing first name');
-        }
-        if (!data.lastName) {
-          throw new BadRequest('Missing last name');
-        }
+    const nameFields = (() => {
+      if (notConfirmedYet && event.nameQuestion) {
+        const { firstName, lastName } = request.body;
+        if (!firstName) { throw new BadRequest('Missing first name'); }
+        if (!lastName) { throw new BadRequest('Missing last name'); }
+        return { firstName, lastName };
       }
-      if (event.emailQuestion && !data.email) {
-        throw new BadRequest('Missing email');
+      return {};
+    })();
+
+    const emailField = (() => {
+      if (notConfirmedYet && event.emailQuestion) {
+        const { email } = request.body;
+        if (!email) { throw new BadRequest('Missing email'); }
+        return { email };
       }
-    }
+      return {};
+    })();
 
     // Check that all questions are answered with a valid answer
     const newAnswers = questions.map((question) => {
-      const answer = _.find(data.answers, { questionId: question.id })?.answer || '';
+      const answer = (() => {
+        // Parse answer from answers or return an empty string if it doesn't exist
+        if (!request.body.answers) { return ''; }
+        const ans = request.body.answers.find((a) => a.questionId === question.id);
+        return ans ? ans.answer : '';
+      })();
```

Comment:
Same as above, but here we can even do the parsing in a nice "pipeline".

```suggestion
      const answer = request.body.answers
        ?.find((a) => a.questionId === question.id)
        ?.answer
        || '';
```

Reply:
Resolved in 8d88854